session/pingpong/accountant_caller.go

```diff
@@ -96,7 +99,20 @@ func (ac *AccountantCaller) GetConsumerData(id string) (ConsumerData, error) {
 	}
 	var resp ConsumerData
 	err = ac.doRequest(req, &resp)
-	return resp, errors.Wrap(err, "could not request consumer data accountant")
+	if err != nil {
+		return ConsumerData{}, errors.Wrap(err, "could not request consumer data from accountant")
+	}
+
+	valid, err := resp.LatestPromise.isValid(id)
```

Comment:
I suppose it's enough to return err by joining `!valid` to the error case, then there's no need to double-check because the promise is invalid in both cases:
```
err = resp.LatestPromise.isValid(id)
if err != nil {
		return ConsumerData{}, errors.New("accountant sent an invalid promise for the ID")
	}
```